spinta/manifests/xsd/helpers.py

```diff
@@ -0,0 +1,749 @@
+                property_id, prop = self.simple_element_to_property(element)
+                if not from_sequence:
+                    prop["required"] = False
+                properties[property_id] = prop
+        return properties
+
+    def get_text_property(self):
+        return {
+            self.deduplicate('text'): {
+                'type': 'string',
+                'external': {
+                    'name': 'text()'
+                }
+            }}
+
+
+class XSDReader:
+
+    def __init__(self, path, dataset_name):
+        self._path = path
+        self.models: list[XSDModel] = []
+        self.custom_types = {}
+        self._dataset_given_name = dataset_name
+        self._set_dataset_and_resource_info()
+        self.deduplicate = Deduplicator()
+
+    def _extract_custom_types(self, node: _Element):
+        custom_types_nodes = node.xpath(f'./*[local-name() = "simpleType"]')
+        custom_types = {}
+        for type_node in custom_types_nodes:
+            type_name = type_node.get("name")
+            restrictions = type_node.xpath(f'./*[local-name() = "restriction"]')
+            property_type_base = restrictions[0].get("base", "")
+            property_type_base = property_type_base.split(":")[1]
+
+            custom_types[type_name] = {
+                "base": property_type_base
+            }
+        self.custom_types = custom_types
+
+    @staticmethod
+    def get_description(element: etree.Element) -> str:
+        annotation = element.xpath(f'./*[local-name() = "annotation"]')
+        if not annotation:
+            annotation = element.xpath(f'./*[local-name() = "simpleType"]/*[local-name() = "annotation"]')
+        description = ""
+        if annotation:
+            documentation = annotation[0].xpath(f'./*[local-name() = "documentation"]')
+            for documentation_part in documentation:
+                if documentation_part.text is not None:
+                    description = f"{description}{documentation_part.text} "
+        return description.strip()
+
+    @staticmethod
+    def node_is_ref(node: _Element):
+        if node.get("ref"):
+            return True
+        return False
+
+    @staticmethod
+    def is_array(element: _Element) -> bool:
+        max_occurs = element.get("maxOccurs", 1)
+        return max_occurs == "unbounded" or int(max_occurs) > 1
+    
+    def _extract_root(self):
+        if self._path.startswith("http"):
+            document = etree.parse(urlopen(self._path))
+            objectify.deannotate(document, cleanup_namespaces=True)
+            self.root = document.getroot()
+        else:
+            with open(self._path) as file:
+                text = file.read()
+                self.root = etree.fromstring(bytes(text, encoding='utf-8'))
+
+    def _set_dataset_and_resource_info(self):
+        self.dataset_name = os.path.splitext(os.path.basename(self._path))[0]
```

Comment:
`dataset.name` turi būti [mažosiomis raidėmis ir be tašo simbolio](https://atviriduomenys.readthedocs.io/dsa/kodiniai-pavadinimai.html#vardu-erdviu).

Dabar matau tokius pavadinimus kaip

    rc_ntr_klasif_VERTINIMAI.VPK_ZONOS

nors turėtu būti

    rc_ntr_klasif_vertinimai_vpk_zonos

Reikia naudoti [to_dataset_name](https://github.com/atviriduomenys/spinta/blob/6930b1af53fcf2b8e044b8dd797df5218c328c19/spinta/utils/naming.py#L16) funckiją.

Reply:
pataisiau